pkg/cli/config/config.go

```diff
@@ -40,51 +26,54 @@ func getDefaultConfigPath() (string, error) {
 	return configFilePath, nil
 }
 
-// Get Backplane ProxyUrl from config
+// GetBackplaneConfiguration parses and returns the given backplane configuration
 func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
+	var filepath string
 
-	// Check proxy url from the config file
-	filePath, err := GetBackplaneConfigFile()
-	if err != nil {
-		return bpConfig, err
-	}
-
-	if _, err := os.Stat(filePath); err == nil {
-		file, err := os.Open(filePath)
+	// Check if user has explicitly defined backplane config path
+	path, bpConfigFound := os.LookupEnv(info.BACKPLANE_CONFIG_PATH_ENV_NAME)
 
+	if bpConfigFound {
+		filepath = path
+	} else {
+		filepath, err = getConfigFilePath()
 		if err != nil {
-			return bpConfig, fmt.Errorf("failed to read file %s : %v", filePath, err)
+			return bpConfig, err
 		}
+	}
 
-		defer file.Close()
-		decoder := json.NewDecoder(file)
-		bpConfig := BackplaneConfiguration{}
-		err = decoder.Decode(&bpConfig)
+	// Check if the config file exists
+	if _, err = os.Stat(filepath); err == nil {
+		// Load config file
+		viper.SetConfigFile(filepath)
+		viper.SetConfigType("json")
 
-		if err != nil {
-			return bpConfig, fmt.Errorf("failed to decode file %s : %v", filePath, err)
+		if err := viper.ReadInConfig(); err != nil {
+			return bpConfig, err
 		}
 
-		return bpConfig, nil
-
-	} else {
-		// check proxy url from user perssitance HTTPS_PROXY env var
-		proxyUrl, hasEnvProxyURL := os.LookupEnv(info.BACKPLANE_PROXY_ENV_NAME)
-
-		// get backplane URL from BACKPLANE_URL env variables
-		bpURL, hasURL := os.LookupEnv(info.BACKPLANE_URL_ENV_NAME)
-
-		if hasURL {
-			if bpURL == "" {
-				return bpConfig, fmt.Errorf("%s env variable is empty", info.BACKPLANE_URL_ENV_NAME)
-			}
-			bpConfig.URL = bpURL
-			if hasEnvProxyURL {
-				bpConfig.ProxyURL = proxyUrl
-			}
+		if err := viper.Unmarshal(&bpConfig); err != nil {
+			return bpConfig, err
+		}
+	}
 
-			return bpConfig, nil
+	// Check if user has explicitly defined backplane URL
+	// It has higher precedence over the config file
+	bpURL, hasURL := os.LookupEnv(info.BACKPLANE_URL_ENV_NAME)
```

Comment:
viper does support read configs considering environment setup, could you try to see how to integrate this? https://github.com/spf13/viper#working-with-environment-variables

Reply:
I see no value use viper without using this feature.

Reply:
I had this added in another branch, moved it here, thanks